packages/mobile/src/import/ImportWallet.tsx

```diff
@@ -42,105 +42,64 @@ const AVERAGE_SEED_WIDTH = AVERAGE_WORD_WIDTH * 24
 // Estimated number of lines needed to enter the account key
 const NUMBER_OF_LINES = Math.ceil(AVERAGE_SEED_WIDTH / Dimensions.get('window').width)
 
-interface State {
-  keyboardVisible: boolean
-  backupPhrase: string
-}
-
-interface DispatchProps {
-  importBackupPhrase: typeof importBackupPhrase
-  hideAlert: typeof hideAlert
-}
-
-interface StateProps {
-  isImportingWallet: boolean
-  connected: boolean
-  isRecoveringFromStoreWipe: boolean
-  accountToRecoverFromStoreWipe: string | undefined
-}
-
-type OwnProps = StackScreenProps<StackParamList, Screens.ImportWallet>
-
-type Props = StateProps & DispatchProps & WithTranslation & OwnProps
-
-const mapStateToProps = (state: RootState): StateProps => {
-  return {
-    isImportingWallet: state.imports.isImportingWallet,
-    connected: isAppConnected(state),
-    isRecoveringFromStoreWipe: recoveringFromStoreWipeSelector(state),
-    accountToRecoverFromStoreWipe: accountToRecoverSelector(state),
-  }
-}
+type Props = StackScreenProps<StackParamList, Screens.ImportWallet>
 
 /**
  * Component shown to users when they are onboarding to the application through the import / recover
  * wallet flow. Allows the user to input their mnemonic phrase to instantiate the account.
  */
-export class ImportWallet extends React.Component<Props, State> {
-  static navigationOptions = {
-    ...nuxNavigationOptions,
-    headerLeft: () => (
-      <TopBarTextButtonOnboarding
-        title={i18n.t('global:cancel')}
-        // Note: redux state reset is handled by UseBackToWelcomeScreen
-        onPress={() => navigate(Screens.Welcome)}
-      />
-    ),
-    headerTitle: () => (
-      <HeaderTitleWithSubtitle
-        title={i18n.t('nuxNamePin1:importIt')}
-        subTitle={i18n.t('onboarding:restoreAccountSteps', { step: '3' })}
-      />
-    ),
-  }
+function ImportWallet({ navigation, route }: Props) {
+  const [backupPhrase, setBackupPhrase] = useState('')
+  const [keyboardVisible, setKeyboardVisible] = useState(false)
+
+  const isImportingWallet = useTypedSelector((state) => state.imports.isImportingWallet)
+  const appConnected = useSelector(isAppConnected)
+  const isRecoveringFromStoreWipe = useTypedSelector(recoveringFromStoreWipeSelector)
+  const accountToRecoverFromStoreWipe = useTypedSelector(accountToRecoverSelector)
 
-  state = {
-    backupPhrase: '',
-    keyboardVisible: false,
+  const dispatch = useDispatch()
+  const { t } = useTranslation(Namespaces.onboarding)
+
+  async function autocompleteSavedMnemonic() {
+    if (!accountToRecoverFromStoreWipe) {
+      return
+    }
+    const mnemonic = await getStoredMnemonic(accountToRecoverFromStoreWipe)
+    if (mnemonic) {
+      setBackupPhrase(mnemonic)
+      onPressRestore()
+    }
   }
-  componentDidMount() {
+
+  React.useEffect(() => {
```

Comment:
Nit: can you please import `useEffect` directly? That's the style that we are trying to standardize on